app/components/header/index.test.tsx

```diff
@@ -0,0 +1,29 @@
+import { fireEvent, render, screen, waitFor } from '@testing-library/react'
+import Header from './index'
+
+jest.mock('next/navigation', () => ({
+  useRouter: jest.fn(),
+}))
+
+describe('Header component: test cases', () => {
+  test('header components element should be rendered', () => {
+    render(<Header />)
+    const buttonEl = screen.getByRole('button')
+    expect(buttonEl).toBeInTheDocument()
+  })
+
+  test('button text should be new todo', () => {
+    render(<Header />)
+    const buttonEl = screen.getByRole('button')
+    expect(buttonEl).toHaveTextContent('+ New Todo')
+  })
+
+  test.skip('new todo button click should re-direct to /new-item route', async () => {
```

Comment:
Why are we using test.skip?

Reply:
That time it was not working. Updated the test case now.